### ISSUE TITLE
How to set MapboxDirections language for instruction via JavaScript?

### ISSUE THREAD

**Issue by KrisKodira** [mapbox/mapbox-gl-directions#209]
I'm using Mapbox directions v4.0.0 inside a HTML file.

I've successfully set the map language to German and the geolocator language is also German but I still get English instructions.
I saw that the language of the instructions get set to English inside the `compile` function but when I want to override the function and I log the parameter that the function receives I only get `en` and I can't do anything with it.

This is my JS:
```
var map = new mapboxgl.Map({
	container: 'map',
	style: 'CUSTOM_STYLE',
	center: [LONG, LAT],
	zoom: 13
});

 
var directions = new MapboxDirections({
	accessToken: mapboxgl.accessToken,
	unit: "metric",
	interactive: false,
	profile: "mapbox/driving",
	alternatives: true,
	compile: function(c){
		console.log(c)
	},
	controls: {
		profileSwitcher: false
	},
	geocoder: {
		language: "de"
	},
	placeholderDestination: "Bitte geben Sie Ihr Ziel ein.",
	placeholderOrigin: "Bitte geben Sie Ihren Startpunkt ein."
});
```

If you need more information just ask and I will provide everything.

**Comment by mourner** [comment]
Thanks for the report! I moved your issue to the more relevant https://github.com/mapbox/mapbox-gl-directions repo.

**Comment by DevIntact** [comment]
It seems the _language_ parameter is not taken in account in https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-directions/v4.0.0/mapbox-gl-directions.js

As we can see, the _language_ parameter is not mentioned : 

```
/**
 * The Directions control
 * @class MapboxDirections
 *
 * @param {Object} options
 * @param {Array} [options.styles] Override default layer properties of the [directions source](https://github.com/mapbox/mapbox-gl-directions/blob/master/src/directions_style.js). Documentation for each property are specified in the [Mapbox GL Style Reference](https://www.mapbox.com/mapbox-gl-style-spec/).
 * @param {String} [options.accessToken=null] Required unless `mapboxgl.accessToken` is set globally
 * @param {String} [options.api="https://api.mapbox.com/directions/v5/"] Override default routing endpoint url
 * @param {Boolean} [options.interactive=true] Enable/Disable mouse or touch interactivity from the plugin
 * @param {String} [options.profile="mapbox/driving-traffic"] Routing profile to use. Options: `mapbox/driving-traffic`, `mapbox/driving`, `mapbox/walking`, `mapbox/cycling`
 * @param {Boolean} [options.alternatives=false] Whether to enable alternatives.
 * @param {Boolean} [options.congestion=false] Whether to enable congestion along the route line.
 * @param {String} [options.unit="imperial"] Measurement system to be used in navigation instructions. Options: `imperial`, `metric`
 * @param {Function} [options.compile=null] Provide a custom function for generating instruction, compatible with osrm-text-instructions.
 * @param {Object} [options.geocoder] Accepts an object containing the query parameters as [documented here](https://www.mapbox.com/api-documentation/#search-for-places).
 * @param {Object} [options.controls]
 * @param {Boolean} [options.controls.inputs=true] Hide or display the inputs control.
 * @param {Boolean} [options.controls.instructions=true] Hide or display the instructions control.
 * @param {Boolean} [options.controls.profileSwitcher=true] Hide or display the default profile switch with options for traffic, driving, walking and cycling.
 * @param {Number} [options.zoom=16] If no bbox exists from the geocoder result, the zoom you set here will be used in the flyTo.
 * @param {String} [options.placeholderOrigin="Choose a starting place"] If set, this text will appear as the placeholder attribute for the origin input element.
 * @param {String} [options.placeholderDestination="Choose destination"] If set, this text will appear as the placeholder attribute for the destination input element.
 * @param {Boolean} [options.flyTo=true] If false, animating the map to a selected result is disabled.
 * @example
 * var MapboxDirections = require('../src/index');
 * var directions = new MapboxDirections({
 *   accessToken: 'YOUR-MAPBOX-ACCESS-TOKEN',
 *   unit: 'metric',
 *   profile: 'mapbox/cycling'
 * });
 * // add to your mapboxgl map
 * map.addControl(directions);
 *
 * @return {MapboxDirections} `this`
 */
```

Here is my code : 

```
map.addControl(new MapboxDirections({
								accessToken: mapboxgl.accessToken,
								profile: 'mapbox/driving',
								alternatives: true,
								unit: 'metric',
								language: 'fr',
								steps: true,
								geocoder: {
									language: 'fr'
								},
								placeholderOrigin: 't\'es ici'
							}), 'top-left');
```

**Comment by shamalli** [comment]
This would be excellent feature of directions addon if we could translate UI strings.

**Comment by DevIntact** [comment]
@KrisKodira @mourner @shamalli @zugaldia : if you need it now, you can download and use this file [https://github.com/DevIntact/mapbox-gl-directions-w-language](url)

It is actually the same version of https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-directions/v4.0.0/mapbox-gl-directions.js but with the  _langage_ parameter which works

I don't know if someone will someday have a look on the pull requests dating since 2 years...

**Comment by geografa** [comment]
Fixed in #222 

Usage:
```
map.addControl(new MapboxDirections({
    accessToken: mapboxgl.accessToken,
    language: 'fr'
}), 'top-left');
```

**Comment by Simbaclaws** [comment]
Apparently when changing the language, the approximate time for how long the direction takes doesn't use the notation of the current language.

What I mean by this is that:

`1**h** 22min`

Could in a different language be:

`1**u** 22min`

The time notation seems to always be english.
Is there a way to fix this without having to change the time notation ourselves after the fact with javascript?
